rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/charts/DiskPieChart.java

```diff
@@ -43,9 +44,45 @@
  */
 public class DiskPieChart extends MBeansChart {
 
+    private int total = 0;
+
     public DiskPieChart(RMController controller, String jmxServerUrl) {
         super(controller, jmxServerUrl, "sigar:Type=FileSystem,Name=*", new String[] { "Total" }, "File System, Mb");
         setYAxesTicksSuffix(" Mb");
+        setTooltipItemHandler(new Function<String, String>() {
+            @Override
+            public String apply(String s) {
+                int indexOfSpace = s.lastIndexOf(" ");
+                String firstHalf = s.substring(0, indexOfSpace);
+                String secondHalf = s.substring(indexOfSpace + 1);
+
+                if (secondHalf.contains(".")) {
+                    secondHalf = secondHalf.substring(0, secondHalf.indexOf("."));
+                }
+
+                long valueInMb = Long.parseLong(secondHalf);
+
+                long percentage = (long) (((double) valueInMb / total) * 100);
+
+                secondHalf = putCommasEveryThreeDigits(secondHalf);
+
+                return firstHalf + " " + secondHalf + " Mb (" + percentage + "%)";
+            }
+        });
+    }
+
+    public static String putCommasEveryThreeDigits(String number) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(number.charAt(0));
+        for (int i = 1; i < number.length(); ++i) {
+
+            if ((number.length() - i) % 3 == 0) {
```

Comment:
Difficult to read here either. Can you make "i" go by increments of 3, a comment would add too?